docs/core/whats-new/dotnet-9/runtime.md

```diff
@@ -106,9 +118,82 @@ static int Sum(int[] arr)
 
 The index variable, `i`, is 4 bytes in size. At the assembly level, 64-bit registers are typically used to hold array indices on x64, and in previous .NET versions, the compiler generated code that zero-extended `i` to 8 bytes for the array access, but continued to treat `i` as a 4-byte integer elsewhere. However, extending `i` to 8 bytes requires an additional instruction on x64. With IV widening, the 64-bit JIT compiler now widens `i` to 8 bytes throughout the loop, omitting the zero extension. Looping over arrays is very common, and the benefits of this instruction removal quickly add up.
 
+#### Post-indexed addressing on Arm64
+
+Index variables are frequently used to read sequential regions of memory. Consider the idiomatic `for` loop:
+
+```csharp
+int sum = 0;
+int[] nums = [0..12];
+
+for (int i = 0; i < nums.Length; i++)
+{
+    sum += nums[i];
+}
+```
+
+For each iteration of the loop, the index variable `i` is used to read an integer in `nums`, and then `i` is incremented. In Arm64 assembly, these two operations look as follows:
+
+```al
+ldr w0, [x1]
+add x1, x1, #4
+```
+
+`ldr w0, [x1]` loads the integer at the memory address in `x1` into `w0`; this corresponds to the access of `nums[i]` in the source code. Then, `add x1, x1, #4` increases the address in `x1` by four bytes, moving to the next integer in `nums`. This instruction corresponds to the `i++` operation executed at the end of each iteration.
+
+Arm64 supports post-indexed addressing, where the "index" register is automatically incremented after its address is used. This means that two instructions can be combined into one, making the loop more efficient. The CPU only needs to decode one instruction instead of two, and the loop's code is now more cache-friendly.
+
+Here's what the updated assembly looks like:
+
+```al
+ldr w0, [x1], #0x04
+```
+
+The `#0x04` at the end means the address in `x1` is incremented by four bytes after it's used to load an integer into `w0`. The 64-bit compiler now uses post-indexed addressing when generating Arm64 code. While x64 doesn't support post-indexed addressing, [induction variable widening](#induction-variable-widening) is similarly useful for optimizing memory accesses with loop-index variables.
+
+#### Strength reduction
+
+Strength reduction is a compiler optimization where an operation is replaced with a faster, logically equivalent operation. This technique is especially useful for optimizing loops. Consider the example `for` loop code shown in the [previous section](#post-indexed-addressing-on-arm64). The following x64 assembly code shows a snippet of the code that's generated for the loop's body:
```

Comment:
> Consider the example `for` loop code shown in the [previous section]

I think (for such a small sample) that this is an anti-pattern. It doesn't strike me as a service to the reader. I hope that someone shares a deep link to this section (because it is interesting). Folks who arrive here would be best served by seeing the example again.

Reply:
I'm ok with copying it in-line. Do you want me to push updates directly to this branch, or do you want to handle edits?

Reply:
I'll handle it - thanks.